backend/src/apiserver/server/task_server.go

```diff
@@ -40,6 +41,19 @@ func (s *TaskServer) validateCreateTaskRequest(request *api.CreateTaskRequest) e
 	if task.GetPipelineName() == "" {
 		return errMustSpecify("PipelineName")
 	}
+	if task.GetNamespace() == "" {
+		return errMustSpecify("Namespace")
+	}
```

Comment:
this PR will require a coordinated release with SDK, because 1.7.0 SDK launchers will no longer work.

I think we can go with this too, but it might be slightly better to keep namespace as optional, because usually people will not expect 1.7.1 SDK works with the backend, but 1.7.0 SDK doesn't. We can change it to be required after SDK 1.8.0

Reply:
Why it won't work with sdk?  Launcher is the only caller of such apis.

Reply:
done